src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ModuleInitializerAttributeShouldNotBeUsedInLibrariesTests.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.ModuleInitializerAttributeShouldNotBeUsedInLibraries,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.ModuleInitializerAttributeShouldNotBeUsedInLibraries,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class ModuleInitializerAttributeShouldNotBeUsedInLibrariesTests
+    {
+        [Theory]
+        [InlineData("public", "public", false)]
+        [InlineData("public", "internal", false)]
+        [InlineData("internal", "public", false)]
+        [InlineData("internal", "internal", false)]
+        [InlineData("public", "public", true)]
+        [InlineData("public", "internal", true)]
+        [InlineData("internal", "public", true)]
+        [InlineData("internal", "internal", true)]
+        public async Task CA2255ModuleInitializerOnMethod(string classModifier, string methodModifier, bool useAsync = false)
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                    Sources = {
+                        @$"
+                        {classModifier} class CA2255ModuleInitializerOnMethod
+                        {{
+                            [{{|CA2255:System.Runtime.CompilerServices.ModuleInitializer|}}]
```

Comment:
Simpler markup syntax.

```suggestion
                            [[|System.Runtime.CompilerServices.ModuleInitializer|]]
```

Reply:
Since the tests (have to) illustrate places where `CS88xx` diagnostics show up, I wanted to explicitly illustrate where `CA2255` is raised to be clear which diagnostic shows up in which tests and avoid a false positive match on the wrong diagnostic.

Reply:
@jeffhandley I think `[| |]` isn't supposed to match CS88xx, it matches the supported diagnostic from the analyzer under test, and (by default) will fail if the analyzer supports multiple diagnostics or if the diagnostic is from the compiler. but I'm not opposing leaving it as is. It's more of style.

Reply:
Ah, I found that I can specify `CompilerDiagnostics = CompilerDiagnostics.None` to disable those diagnostics from showing up in the verification.